tests/unit/App.spec.js

```diff
@@ -89,4 +89,55 @@ Given("an App", () => {
       expect(app.vm.url).to.equal(expectedUrl);
     });
   });
+
+  When("a response is returned", () => {
+    beforeEach(async () => {
+      app.setData({
+        wasResponseAnError: false,
+        statusText: ""
+      });
+
+      app.setMethods({
+        request: app.vm.handleResponse
+      });
+    });
+
+    And("the response was a 4xx", () => {
+      const response = {
+        status: 404,
+        statusText: "Not Found"
+      };
+
+      beforeEach(async () => {
+        app.vm.request(response);
+      });
+
+      Then("the response was an error", () => {
+        expect(app.vm.wasResponseAnError).to.equal(true);
+      });
+
+      Then("the status text is set correctly", () => {
+        expect(app.vm.statusText).to.equal("Not Found");
+      });
+    });
+
+    And("the response was a 2xx", () => {
+      const response = {
+        status: 200,
+        statusText: "OK"
+      };
+
+      beforeEach(async () => {
+        app.vm.request(response);
+      });
+
+      Then("the response was not an error", () => {
+        expect(app.vm.wasResponseAnError).to.equal(false);
+      });
+
+      Then("the status text is not set", () => {
+        expect(app.vm.statusText).to.equal("");
```

Comment:
Similar thing here. You're just testing the implementation. You should be testing the overall output

Reply:
Done